### PR TITLE
Newyolo

### PR DESCRIPTION
**Pull request by jakepoz**
1. New Yolo v5s Onnx files with higher accuracy
2. Easy export of ONNX file intermediate layers
2. Non max suppression of bounding boxes
3. Unit tests for reward function changes.

### CODE REVIEW

File: tests/test_bag_brain_io.py [hunk 1]
```diff
@@ -7,14 +7,14 @@
...
-    test_onnx = "/home/jake/bagfiles/test_bags/freshplant18200.onnx"
+    test_bag = "/home/jake/bagfiles/test_bags/record-brain-sac-pleasant-frog-450-02048-samp0.0_2021-09-06-13-24-24_3.bag"
+    test_onnx = "/home/jake/hipow-checkpoints/sac-pleasant-frog-450-02048.onnx"
     test_yolo_onnx = os.path.join(os.path.dirname(__file__), "..", "yolov5s.onnx")
 
     def test_lstm_sample(self):
         bag = rosbag.Bag(self.test_bag, "r")
         onnx = rt.InferenceSession(self.test_onnx)
-        history_size = 8
+        history_size = 240
```
Comment (lostmsu) [inline tests/test_bag_brain_io.py:17]: =O

File: train.py [hunk 2]
```diff
@@ -270,8 +268,7 @@ def write_bag_cache(bag_file: str, bag_cache_path: str, backbone_onnx_path: str,
     parser.add_argument('--epoch-steps', type=int, default=100, help='how often to save checkpoints')
     parser.add_argument('--seed', type=int, default=None, help='training seed')
     parser.add_argument('--lstm-history', type=int, default=240, help='max amount of prior steps to feed into a network history')
-    parser.add_argument('--history-indexes', type=str, default='-1,-2,-3,-5,-8,-13,-21,-34,-55,-89,-144,-233',
-                        help='which indexes to pass into the network')
+    parser.add_argument('--history-indexes', type=str, default='-1,-2,-3,-5', help='which indexes to pass into the network')
```
Comment (lostmsu) [inline train.py:271]: Leftover from testing?

**Review by lostmsu — COMMENTED** [review]
Considering it is Friday evening, I don't see any issues.

File: train.py (hunk 2 above)
Comment (jakepoz) [reply train.py:271]: No, decided to just go down to 1,2,3,5 on purpose, not the huge history of the fibonacci tests.

File: tests/test_bag_brain_io.py (hunk 1 above)
Comment (lostmsu) [reply tests/test_bag_brain_io.py:17]: Can you explain this change?

**Comment by lostmsu** [comment]
Considering training went well, I think this can be merged.

File: tests/test_bag_brain_io.py (hunk 1 above)
Comment (jakepoz) [reply tests/test_bag_brain_io.py:17]: When you export ONNX, you need to pass sample tracing data into it that includes at least as many elements as you reference.

**Comment by jakepoz** [comment]
Kk, merging .

**Pull request merged**